sys/sys/user.h

```diff
@@ -524,7 +526,8 @@ struct kinfo_vmentry {
 	uint16_t kve_status;			/* Status flags. */
 	uint64_t kve_vn_fsid;			/* dev_t of vnode location */
 	uint64_t kve_vn_rdev;			/* Device id if device. */
-	int	 _kve_ispare[8];		/* Space for more stuff. */
+	uint64_t kve_reservation;		/* Map reservation */
+	int	 _kve_ispare[6];		/* Space for more stuff. */
```

Comment:
Ick, the padding varies between ILP64 and LP64?  Not your fault, but I think this means that we no longer occupy the size on at least one of the two, no?

Reply:
Why would it vary? There's no padding in the structure up until this point, all the 16/32s have been carefully grouped together, and so we replace an aligned pair of ints with a uint64_t, thereby leaving the structure layout unaltered?

Reply:
Sorry, I didn't mean inter-field padding, I meant the "_kve_ispare" field's size.

Reply:
It’s 6 ints on both? What am I missing?..

Reply:
Changing from 8 ints to 1 uint64_t + 6 ints will be the same size on a LP64 machine w/ 32 bit ints, but will shrink the structure by 8 bytes on ILP64 w/ 64 bit ints, right?  But I guess there aren't any FreeBSD ILP64 systems, only LP64?

I was, I think, just expressing surprise that the ispare field was measured in ints and not in one of the C99 inttypes.

Reply:
IIRC one of the lessons people learned from Cray building an ILP64 Unix was that you really don't want to do that.